pjmedia/src/pjmedia-codec/and_vid_mediacodec.cpp

```diff
@@ -0,0 +1,1923 @@
+                                      const pjmedia_vid_codec_info *info,
+                                      pjmedia_vid_codec_param *attr )
+{
+    unsigned i;
+
+    PJ_ASSERT_RETURN(factory == &and_media_factory.base, PJ_EINVAL);
+    PJ_ASSERT_RETURN(info && attr, PJ_EINVAL);
+
+    for (i = 0; i < PJ_ARRAY_SIZE(and_media_codec); ++i) {
+        if (and_media_codec[i].enabled && info->pt != 0 &&
+            (info->fmt_id == and_media_codec[i].fmt_id))
+        {
+            break;
+        }
+    }
+
+    if (i == PJ_ARRAY_SIZE(and_media_codec))
+        return PJ_EINVAL;
+
+    pj_bzero(attr, sizeof(pjmedia_vid_codec_param));
+
+    attr->dir = PJMEDIA_DIR_ENCODING_DECODING;
+    attr->packing = PJMEDIA_VID_PACKING_PACKETS;
+
+    /* Encoded format */
+    pjmedia_format_init_video(&attr->enc_fmt, info->fmt_id,
+                              DEFAULT_WIDTH, DEFAULT_HEIGHT, DEFAULT_FPS, 1);
+
+    /* Decoded format */
+    pjmedia_format_init_video(&attr->dec_fmt, PJMEDIA_FORMAT_I420,
+                              DEFAULT_WIDTH, DEFAULT_HEIGHT, DEFAULT_FPS, 1);
+
+    attr->dec_fmtp = and_media_codec[i].dec_fmtp;
+
+    /* Bitrate */
+    attr->enc_fmt.det.vid.avg_bps = DEFAULT_AVG_BITRATE;
+    attr->enc_fmt.det.vid.max_bps = DEFAULT_MAX_BITRATE;
+
+    /* Encoding MTU */
+    attr->enc_mtu = PJMEDIA_MAX_VID_PAYLOAD_SIZE;
+
+    return PJ_SUCCESS;
+}
+
+static pj_bool_t codec_exists(const pj_str_t *codec_name)
+{
+    AMediaCodec *codec;
+    char *codec_txt;
+
+    codec_txt = codec_name->ptr;
+
+    codec = AMediaCodec_createCodecByName(codec_txt);
+    if (!codec) {
+	PJ_LOG(4, (THIS_FILE, "Failed creating codec : %.*s", codec_name->slen,
+		   codec_name->ptr));
+	return PJ_FALSE;
+    }
+    AMediaCodec_delete(codec);
+
+    return PJ_TRUE;
+}
+
+void add_codec(struct and_media_codec *codec,
+	       unsigned *count, pjmedia_vid_codec_info *info)
+{
+    info[*count].fmt_id = codec->fmt_id;
+    info[*count].pt = codec->pt;
+    info[*count].encoding_name = pj_str((char *)codec->name);
+    info[*count].encoding_desc = pj_str((char *)codec->description);
+
+    info[*count].clock_rate = 90000;
+    info[*count].dir = PJMEDIA_DIR_ENCODING_DECODING;
+    info[*count].dec_fmt_id_cnt = 1;
+    info[*count].dec_fmt_id[0] = PJMEDIA_FORMAT_I420;
+    info[*count].packings = PJMEDIA_VID_PACKING_PACKETS;
+    info[*count].fps_cnt = 3;
+    info[*count].fps[0].num = 15;
+    info[*count].fps[0].denum = 1;
+    info[*count].fps[1].num = 25;
+    info[*count].fps[1].denum = 1;
+    info[*count].fps[2].num = 30;
+    info[*count].fps[2].denum = 1;
+    ++*count;
+}
+
+static void get_codec_name(pj_bool_t is_enc,
+	                   pj_bool_t prio,
+			   pjmedia_format_id fmt_id,
+			   pj_str_t **codec_name,
+			   unsigned *codec_num)
+{
+    pj_bool_t use_sw_enc = PJMEDIA_AND_MEDIA_PRIO_SW_VID_ENC;
+    pj_bool_t use_sw_dec = PJMEDIA_AND_MEDIA_PRIO_SW_VID_DEC;
+
+    *codec_num = 0;
+
+    switch (fmt_id) {
+
```

Comment:
Looks like an additional compile-time check required here to avoid `use of undeclared identifier 'H264_sw_encoder'` compilation error.
```suggestion
#if PJMEDIA_HAS_AND_MEDIA_H264
```
The same thing for other codecs